### DISCUSSION TITLE
Authentication user with SSO

### DISCUSSION THREAD

**Discussion by liratanak** [outline/outline#1592] [Self Hosting] [answered]
We do have system which team using internally. Without Google Account or Slack, any possible options like we could do integration or auto-login by SSO from a logged in user our internal system ?

I have checked https://www.getoutline.com/developers but I didn't found what I'm looking for, please advise, thanks you

**Comment by tommoor** [comment, marked as answer]
> any possible options like we could do integration or auto-login by SSO from a logged in user our internal system ?

It would be possible at some point (on the roadmap), but it doesn't exist today – we're focused on signin with Google Workspace and Slack.

**Reply by houtbyte** [reply]
Where is documented, how to login with Google Workspace?

**Reply by liratanak** [reply]
@houtbyte - in [.env.sample](https://github.com/outline/outline/blob/develop/.env.sample) all just explained there.

**Reply by reyman** [reply]
Lots of university use ldap or Shiboleth (like mine) if you are interested by this type of public.

**Comment by IgorYarovenko** [comment]
How I can add  custom Authentication user with SSO? Or it doesn't exist today?  thanks

**Reply by tommoor** [reply]
Could you be more specific? What is "custom authentication user"

**Reply by IgorYarovenko** [reply]
We have auth system which team using internally, without Google Account , Slack or Azure.
Can I could do integration or auto-login by SSO from a logged in user our internal system?

**Reply by tommoor** [reply]
You could build any authentication provider, yes. Of course you would need to fork the codebase to do this.

https://github.com/outline/outline/blob/main/server/auth/providers/README.md

**Reply by IgorYarovenko** [reply]
Great. Thanks
